### PR TITLE
fix: DHIS2-10945: Sorting issue causing invalid SQL (2.36)

### PR DESCRIPTION
**Pull request by maikelarabori**
When fixing and investigating this issue I found out we had a general issue with the "sorting" of results. Please, see the details below.

**The issue:**
This endpoint is querying six different entities all at once (from different tables and multiple joins), consolidating the results, sorting and paging. The sorting stage was causing issues because it was being done on the DB side and also on the Java side (to try to keep the same sorting of existing endpoints).
But the sort algorithm on the DB is different from Java, and it can even vary from platform to platform (discovered after some research). So when the DB results were consolidated we were re-sorting on the Java side, which was causing some items to disappear or being showed twice during the pagination.

**Solution:**
Instead of sending isolated queries, consolidating the results and sorting, we are now querying the entities and sorting ALL together only at the DB level. This avoids re-sorting in Java and we'll rely entirely on the DB. Because the DB is the main source of querying, sorting and consolidation (through UNION) we won't see problems during the pagination.

**Main changes:**
I removed the sorting layer on the Java side and consolidated all queries + sorting at the DB level.
During this process, I needed to apply some refactoring which also opened a window for small code clean-ups.

**_Drawback_**
As other existing endpoints use in-memory Java sorting and pagination, there will be a slight difference in sorting between this endpoint and the existing ones. As exposed above, this endpoint will fully rely on DB sorting. There is no way to fix it unless we apply the same behaviour of existing endpoints (in-memory Java sorting and pagination), which would cause performance issues for large DBs.

**_The changes ONLY AFFECT the /dataItems endpoint. No other part of the application will be impacted._**

### CODE REVIEW

**Review by stian-sandvold — APPROVED** [review]
Approved, pending @larshelge 's approval

**Comment by Philip-Larsen-Donnelly** [comment]
I'm going to deploy a build of this PR branch to test before merging.

**Review by Philip-Larsen-Donnelly — APPROVED** [review]

**Pull request merged**
